resolwe/flow/migration_ops.py

```diff
@@ -0,0 +1,373 @@
+            )
+
+        if not migrated_processes:
+            return
+
+        # Migrate all data objects.
+        data = Data.objects.filter(process__in=migrated_processes)
+        self.migrate_data(data, from_state)
+
+        # Update data migration log.
+        for datum in data:
+            DataMigrationHistory.objects.create(
+                migration=self.migration_id,
+                data=datum,
+                metadata=self.describe_data_migration(datum)
+            )
+
+    def database_backwards(self, app_label, schema_editor, from_state, to_state):
+        """Backward migration not possible."""
+        raise NotImplementedError
+
+
+class ResolweProcessAddField(ResolweProcessOperation):  # pylint: disable=abstract-method
+    """Operation, which adds a new field to existing process schema."""
+
+    def __init__(self, process, field, schema, default=None):
+        """Construct a process field add operation.
+
+        :param process: Process slug
+        :param field: Dot-separated field path, starting with the top-level
+            database field name (must be either ``input`` or ``output``)
+        :param schema: Field schema definition
+        :param default: An operation that generates default values, which
+            must be an instance of :class:`DataDefaultOperation` or its
+            subclass
+        """
+        self._raw_field = field
+        self.field = field.split('.')
+        if self.field[0] not in ('input', 'output'):
+            raise ValueError("Field path must start with either input or output")
+        if len(self.field) < 2:
+            raise ValueError("Field path must contain at least two levels")
+        if len(self.field) != 2:
+            # TODO: Support nested fields.
+            raise NotImplementedError("Field path must contain exactly two levels")
+        schema_type = self.field.pop(0)
+
+        self.schema = schema
+        jsonschema.validate([schema], FIELD_SCHEMA)
+
+        if schema['name'] != self.field[-1]:
+            raise ValueError("Field name in schema differs from field path")
+
+        if not default and schema.get('required', True):
+            raise ValueError("A required field must have a default")
+        if default and not isinstance(default, DataDefaultOperation):
+            raise TypeError("Default must be an instance of DataDefaultOperation")
+        self.default = default
+
+        super(ResolweProcessAddField, self).__init__(
+            process=process,
+            schema_type=schema_type,
+            field=field,
+            schema=schema,
+            default=default
+        )
+
+    def deconstruct(self):
+        """Deconstruct operation."""
+        return (
+            self.__class__.__name__,
+            [],
+            {
+                'process': self.process,
+                'field': self._raw_field,
+                'schema': self.schema,
+                'default': self.default,
```

Comment:
`schema_type` is used in `ResolweProcessOperation.__init__`. Probably you have to add it here?

Reply:
No, because in this case it is computed by the constructor from `field`.